src/hooks/useCombinedRefs.ts

```diff
@@ -12,16 +12,27 @@ export function useCombinedRefs<T>(...refs: (ForwardedRef<T> | null | undefined)
   const combinedRef = useRef<T | null>(null)
 
   React.useEffect(() => {
-    for (const ref of refs) {
-      if (!ref) {
-        return
-      }
-      if (typeof ref === 'function') {
-        ref(combinedRef.current ?? null)
-      } else {
-        ref.current = combinedRef.current ?? null
+    const setRefs = (current: T | null = null) => {
```

Comment:
Super nit that you can totally ignore if you want: I have a slight preference for this to be a `function setRefs(...)` instead of the current syntax. Obviously it doesn't make a difference, but I think it improves readability.